modules/Microsoft.Storage/storageAccounts/deploy.bicep

```diff
@@ -338,6 +349,21 @@ module storageAccount_roleAssignments '.bicep/nested_roleAssignments.bicep' = [f
   }
 }]
 
+resource storageAccount_policyExemptions 'Microsoft.Authorization/policyExemptions@2022-07-01-preview' = [for policyExemption in policyExemptions: if (!empty(policyExemptions)) {
```

Comment:
Wouldn't the storage account deployment be already blocked before we even make it to the excemption?

Reply:
it is not, the storage account is deployed. The common tests for both storage accounts and key vaults deploy a policy assignment that deny's creation of a resource that doesn't meet the policy.  I verified that the test is a true test in that the resource doesn't meet the policy, but the exception works and the deployment succeeds.

Reply:
Is the policy a deny or an audit policy?

Reply:
@eriqua, it is a Deny policy (https://portal.azure.com/#blade/Microsoft_Azure_Policy/PolicyDetailBlade/definitionId/%2Fproviders%2FMicrosoft.Authorization%2FpolicyDefinitions%2Fc9d007d0-c057-4772-b18c-01e546713bcd) and in the test deployed with the Deny effect. Same for Key Vault.

Reply:
This is exactly why I needed the PR. We can not create Resource Level exemptions without the resource ID and we can't create the resource with a deny policy in place without the exemption. The only way to do this is through ARM as a single deployment. ARM sorts it out and allows the resource to be created due to the exemption. Without this we would have to temporarily place an exemption at the RG level or disable the policy.